sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/CollectionsResource.java

```diff
@@ -0,0 +1,294 @@
+                                }
+                                """))),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_400),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_401),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+      })
+  @POST
+  public Uni<RestResponse<Object>> createCollection(
+      @Context UriInfo uriInfo,
+      @PathParam("namespace") String namespace,
+      @NotNull(message = "payload not provided") @Valid CreateCollectionDto body) {
+
+    String collection = body.name();
+
+    // go get the existing table
+    return tableManager
+        .getValidCollectionTable(namespace, collection)
+
+        // in case there is a table, ensure we report as error
+        .map(
+            table -> {
+              int code = Response.Status.CONFLICT.getStatusCode();
+              ApiError error =
+                  new ApiError(
+                      "Create failed: collection %s already exists.".formatted(collection), code);
+              return RestResponse.ResponseBuilder.create(code).entity(error).build();
+            })
+
+        // otherwise if table does not exist, table manager would fire the error
+        // with DATASTORE_TABLE_DOES_NOT_EXIST
+        .onFailure(
+            t -> {
+              if (t instanceof ErrorCodeRuntimeException ec) {
+                return Objects.equals(ec.getErrorCode(), ErrorCode.DATASTORE_TABLE_DOES_NOT_EXIST);
+              }
+              return false;
+            })
+        .recoverWithUni(
+            () ->
+                tableManager
+                    .createCollectionTable(namespace, collection)
+                    .map(
+                        created -> {
+                          URI location = uriInfo.getBaseUriBuilder().path(collection).build();
+                          return RestResponse.created(location);
+                        }));
+  }
+
+  @Operation(description = "Delete a collection in a namespace.")
+  @Parameters(
+      value = {
+        @Parameter(name = "namespace", ref = OpenApiConstants.Parameters.NAMESPACE),
+        @Parameter(name = "collection", ref = OpenApiConstants.Parameters.COLLECTION)
+      })
+  @APIResponses(
+      value = {
+        @APIResponse(responseCode = "204", description = "Collection deleted."),
+        @APIResponse(
+            responseCode = "404",
+            description = "Not found.",
+            content =
+                @Content(
+                    examples = {
+                      @ExampleObject(ref = OpenApiConstants.Examples.NAMESPACE_DOES_NOT_EXIST),
+                      @ExampleObject(ref = OpenApiConstants.Examples.COLLECTION_DOES_NOT_EXIST)
+                    },
+                    schema =
+                        @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                            implementation = ApiError.class))),
```

Comment:
This is super cool, we can now add multiple examples for the same status code @polandll @EricBorczuk :

![Screenshot from 2022-05-20 16-51-24](https://user-images.githubusercontent.com/10600041/169554716-2c9442a5-1156-49ff-a6e8-a0244232feab.png)